core/src/main/java/org/apache/iceberg/SnapshotProducer.java

```diff
@@ -290,7 +298,27 @@ public Snapshot apply() {
         operation(),
         summary(base),
         base.currentSchemaId(),
-        manifestList.location());
+        manifestList.location(),
+        lastRowId,
+        addedRows);
+  }
+
+  private Long calculateAddedRows(List<ManifestFile> manifests) {
+    return manifests.stream()
+        .filter(
+            manifest ->
+                manifest.snapshotId() == null
+                    || Objects.equals(manifest.snapshotId(), this.snapshotId))
+        .mapToLong(
+            manifest -> {
+              Preconditions.checkArgument(
+                  manifest.addedRowsCount() != null,
+                  "Cannot determine number of added rows in snapshot because"
+                      + " the entry for manifest %s is missing the field `added-rows-count`",
+                  manifest.path());
```

Comment:
Actually sorry if this is a naive q and missing this earlier, how does this work for manifest rewrites? In that case there would be a new manifest(s) for this snapshot and no new added rows, but this would fail the check currently if I follow right.

Reply:
Ah I checked this out locally and wrote a test...addedRowsCount would just be 0 in that case. Cool